imageio/plugins/tifffile.py

```diff
@@ -0,0 +1,153 @@
+
+
+class TiffFormat(Format):
+
+    """ Provides support for a wide range of Tiff images.
+
+    Parameters for reading
+    ----------------------
+    None
+
+    Parameters for saving
+    ---------------------
+    bigtiff : bool
+        If True, the BigTIFF format is used.
+    byteorder : {'<', '>'}
+        The endianness of the data in the file.
+        By default this is the system's native byte order.
+    software : str
+        Name of the software used to create the image.
+        Saved with the first page only.
+    photometric : {'minisblack', 'miniswhite', 'rgb'}
+            The color space of the image data.
+            By default this setting is inferred from the data shape.
+    planarconfig : {'contig', 'planar'}
+        Specifies if samples are stored contiguous or in separate planes.
+        By default this setting is inferred from the data shape.
+        'contig': last dimension contains samples.
+        'planar': third last dimension contains samples.
+    resolution : (float, float) or ((int, int), (int, int))
+        X and Y resolution in dots per inch as float or rational numbers.
+    description : str
+        The subject of the image. Saved with the first page only.
+    compress : int
+        Values from 0 to 9 controlling the level of zlib compression.
+        If 0, data are written uncompressed (default).
+    volume : bool
+        If True, volume data are stored in one tile (if applicable) using
+        the SGI image_depth and tile_depth tags.
+        Image width and depth must be multiple of 16.
+        Few software can read this format, e.g. MeVisLab.
+    writeshape : bool
+        If True, write the data shape to the image_description tag
+        if necessary and no other description is given.
+    extratags: sequence of tuples
+        Additional tags as [(code, dtype, count, value, writeonce)].
+
+        code : int
+            The TIFF tag Id.
+        dtype : str
+            Data type of items in 'value' in Python struct format.
+            One of B, s, H, I, 2I, b, h, i, f, d, Q, or q.
+        count : int
+            Number of data values. Not used for string values.
+        value : sequence
+            'Count' values compatible with 'dtype'.
+        writeonce : bool
+            If True, the tag is written to the first page only.
+    """
+
+    def _can_read(self, request):
+        if request.filename.lower().endswith(TIFF_FORMATS):
+            return True  # We support any kind of image data
+        else:
+            return False
+
+    def _can_write(self, request):
+        if request.filename.lower().endswith(TIFF_FORMATS):
+            return True  # We support any kind of image data
+        else:
+            return False
+
+    # -- reader
+
+    class Reader(Format.Reader):
+
+        def _open(self):
+            if not tifffile:
+                load_tifffile()
+            self._tf = tifffile.TiffFile(self.request.get_local_filename())
```

Comment:
Does `TiffFile` also accept a file object? It should be faster in some cases. http://imageio.readthedocs.org/en/latest/devapi.html#imageio.core.Request.get_file